src/@next/pages/CheckoutPage/CheckoutPage copy.tsx

```diff
@@ -0,0 +1,398 @@
+      const errors = dataError?.error;
+      setSubmitInProgress(false);
+      if (errors) {
+        setPaymentGatewayErrors(errors);
+        const paymentStepLink = steps.find(
+          step => step.step === CheckoutStep.Payment
+        )?.link;
+        if (paymentStepLink) {
+          push(paymentStepLink);
+        }
+      } else {
+        setPaymentGatewayErrors([]);
+        handleStepSubmitSuccess(CheckoutStep.Review, {
+          id: data?.order?.id,
+          orderNumber: data?.order?.number,
+          token: data?.order?.token,
+        });
+      }
+    } else {
+      setPaymentGatewayErrors([
+        {
+          message: translateAdyenConfirmationError(
+            query.resultCode as string,
+            intl
+          ),
+        },
+      ]);
+      const paymentStepLink = steps.find(
+        step => step.step === CheckoutStep.Payment
+      )?.link;
+      if (paymentStepLink) {
+        push(paymentStepLink);
+        setSubmitInProgress(false);
+        setPaymentConfirmation(false);
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (!metadata) {
+      return;
+    }
+    if (!metadata[CheckoutMetadataTypes.JobFunction]) {
+      const { link } = CHECKOUT_STEPS[0];
+      replace(link);
+      return;
+    }
+    if (!metadata[CheckoutMetadataTypes.JobTitle]) {
+      const { link } = CHECKOUT_STEPS[1];
+      replace(link);
+    }
+  }, [metadata]);
+
+  // useRedirectToCorrectCheckoutStep(cartLoaded);
+  useEffect(() => setSelectedPaymentGateway(payment?.gateway), [
+    payment?.gateway,
+  ]);
+  useEffect(() => setSelectedPaymentGatewayToken(payment?.token), [
+    payment?.token,
+  ]);
+
+  useEffect(() => {
+    const paymentConfirmStepLink = CHECKOUT_STEPS.find(
+      step => step.step === CheckoutStep.PaymentConfirm
+    )?.link;
+    if (
+      !submitInProgress &&
+      checkout &&
+      pathname === paymentConfirmStepLink &&
+      !paymentConfirmation
+    ) {
+      handlePaymentConfirm();
+    }
+  }, [pathname, query, submitInProgress, checkout]);
+  // console.log(activeStep.step);
+
+  useEffect(() => {
+    const createCheckout = async () => {
+      try {
+        const email = user?.email || "customer@example.com";
+        const response = await apolloClient.mutate({
+          mutation: createCheckoutQuery,
+          variables: {
+            lines: (items || []).map(item => ({
+              quantity: item.quantity,
+              variantId: item.variant.id,
+            })),
+            email,
+          },
+        });
+        const newCheckout = response?.data?.checkoutCreate?.checkout;
+        localStorage.setItem(
+          "data_checkout",
+          JSON.stringify({
+            ...newCheckout,
+            lines: items,
+            metadata: {},
+          })
+        );
+        history.go(0);
+      } catch (error) {
+        console.log("Error on creating checkout", error); // eslint-disable-line no-console
```

Comment:
Must the user not be informed as well that something went wrong? Maybe new ticket?

Reply:
Maybe, not sure how can we display this kind of errors for now

Reply:
`CheckoutPage copy.tsx` we don't use this anywhere, Just made a copy for the reference